ample/util/options_processor.py

```diff
@@ -427,28 +406,19 @@ def process_options(optd):
     """
             logger.warn(msg)
             optd['use_shelxe'] = False
-    #
-    # If shelxe_rebuild is set we need use_shelxe to be set
-    #
-    if optd['shelxe_rebuild'] and not optd['use_shelxe']:
-        msg = 'shelxe_rebuild is set but use_shelxe is False. Please make sure you have shelxe installed.'
-        exit_util.exit_error(msg)
-
-    # Output various information to the user
-    logger.info('Running on %d processors' % optd['nproc'])
+    if optd['shelxe_rebuild']:
+        optd['shelxe_rebuild_arpwarp'] = True
+        optd['shelxe_rebuild_buccaneer'] = True
 
-    if optd['make_frags']:
-        if optd['use_homs']:
-            logger.info('Making fragments (including homologues)')
+    # Model building programs
+    if optd['refine_rebuild_arpwarp'] or optd['shelxe_rebuild_arpwarp']:
+        if not (os.environ.has_key('warpbin')
```

Comment:
AAAAAAAAARRRRGH ... don't ever ever ever use `has_key` again!!

Only joking, it's deprecated in Python 3.X and [not really recommended either](https://docs.python.org/3.1/whatsnew/3.0.html#builtins). Turn this into `'warpbin' in os.environ`.